indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -60,15 +63,23 @@ public class KillUnusedSegmentsTask extends AbstractFixedIntervalTask
 {
   private static final Logger LOG = new Logger(KillUnusedSegmentsTask.class);
 
+  // We split this to try and keep each nuke operation relatively short, in the case that either
+  // the database or the storage layer is particularly slow.
+  private static final int DEFAULT_SEGMENT_NUKE_BATCH_SIZE = 100;
+
   private final boolean markAsUnused;
+  private final int batchSize;
+
+  private long countBatchesIssued = 0;
```

Comment:
```suggestion
  private long numBatchesProcessed;
```

Reply:
fixed in d3d5a71ddf41dd258a7721b8bd1893b25e83cd11